interpret.py

```diff
@@ -43,54 +44,137 @@ def getoutput(self):
 
 class DebugFlow:
 	"""
-		DebugFlow contain the logic for ast traversal, detecting what object to
-		render and what subexpression to display.
+	DebugFlow contain the logic for ast traversal, detecting what object to
+	render and what subexpression to display.
 	"""
 	def __init__(self, interp = None):
 		self.interpreter = interp if interp else Interpreter()
 		self.interpreter.getoutput() # get out the python notice
+		self.seen = set()
+		self.stream = print
```

Comment:
Q: Shouldn't the print come as an argument from the parent.
Maybe have something like
```
def __init__(self, interp = None, stream = print):
    ....
    self.stream = stream
```